File: algorithms/sorting/bubbleSort/bubble-sort.c
#include <stdio.h>

void bubbleSort(int array[], int size) {
    for (int i = 0; i < size - 1; i++) {
        int f = 0;
        for (int j = 0; j < size - i - 1; j++) {
            if (array[j] < array[j + 1]) {
                int tmp = array[j];
                array[j] = array[j + 1];
                array[j + 1] = tmp;
                f = 1; 
            }
        }
        if (!f) break;
    }
}

void printArr(int array[], int size) {
    for (int i = 0; i < size; i++) {
        printf("%d, ", array[i]);
    }
    printf("\n");
}

int main() {
    int arr[] = {64, 54, 25, 64, 22, 88, 88};
    int size = sizeof(arr) / sizeof(arr[0]);

    bubbleSort(arr, size);
    printArr(arr, size);

    return 0;
}

File: algorithms/sorting/bubbleSort/bubble-sort.js
export function bubbleSort(array, compareFn = (a, b) => a - b) {
    const size = array.length;
    for(let i = 0; i < size - 1; ++i) {
        let f = false;

        for(let j = 0; j < size - i - 1; ++j) {
            if (compareFn(array[j], array[j + 1]) > 0) {
                [array[j], array[j + 1]] = [array[j + 1], array[j]];
                f = true;
            }
        }
        if (!f) {
            break;
        }
    }
    return array;
}